src/main/_docs/workspace-administration/ws-data-model-workspaces.md

```diff
@@ -0,0 +1,196 @@
+  "config"       : {},      // Configuration of this workspace instance
+  "runtime"      : {},      // The runtime of this workspace instance
+  "attributes"   : {}       // The workspace attributes
+}
+```
+Possible `status` values are: `STARTING`,`RUNNING`, `SNAPSHOTTING`, `STOPPING` and `STOPPED`. 
+
+
+## WorkspaceConfig Object
+
+WorkspaceConfig JSON:
+
+```json  
+"workspaceConfig": {
+  "name"           : STRING,    // The name of this workspace
+  "description"    : STRING,    // The workspace description
+  "defaultEnv"     : STRING,    // The name of env that powers this workspace
+  "environments"   : {},        // Map of runtime envs this workspace uses
+  "projects"       : [{}],      // List of projects included in the workspace
+  "commands"       : [{}]       // List of commands that build & run projects
+}
+```
+
+Every workspace can have one or more environments which are used to run the code against a stack of technology. Every workspace has exactly one environment which acts as a special "development environment", for which projects are synchronized into and developer services are injected, such as intellisense, workspace agents, SSH, and plug-ins.  
+
+Set `defaultEnv` to the name of the environment that should act as the Docker-powered environment that powers the workspace when it boots. This name must match the name given to an object in the `environments` array. Che will create a container off of this environment when the workspace is launched. 
+
+## Environments 
+Each environments are constructed of one or more machines, each one is an individual container. An environment can be comprised of multiple machines that are linked together, such as when you want a database running on a different machine than your debugger.
+```json  
+{
+  "environments": {
+    "name": {
+      "machines" : {}, // Map of machines configurations
+      "recipe"   : {}  // The recipe (the main script) to define this environment
+    }
+  }  
+}
+```
+          
+### Recipe object 
+
+```json  
+ {
+   "recipe" : {
+     "contentType" : STRING, //  Content type of the environment recipe
+     "type"        : STRING, //  Type of the environment
+     "content"     : STRING, //  Content of an environment recipe
+     "location"    : STRING  //  Location of an environment recipe
+   }
+ }
+```
+ Content and location fields are mutually exclusive, i.e. only one can be present.
+ The source of a machine configuration object is supporting several types when using `docker` as machine configuration type, here are the supported source options
+ 
+ #### dockerfile type
+ It provides a docker runtime. Link to the Dockerfile recipe can be provided by a link, using `location` field or by providing directly the content of the Dockerfile, using `content `field
+```json 
+ {
+   "recipe": [
+     {
+       "content"    : STRING,
+       "type"       : "dockerfile",
+       "contentType": "application/x-dockerfile"
+     }
+   ]
+ }
+```
```

Comment:
We need to add concrete samples, as here: 
https://www.eclipse.org/che/docs/workspace/data-model-stacks/index.html#dockerfile-type

BTW - does it means we have inconsistency in the format?